server/metric/mock_metric_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"context"
+	"log"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/genproto/googleapis/monitoring/v3"
+	metrics "google.golang.org/genproto/googleapis/api/metric"
+	mockMetricServer "github.com/googleinterns/cloud-operations-api-mock/server/metric"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+	"google.golang.org/protobuf/proto"
+)
+
+const bufSize = 1024 * 1024
+
+var (
+	client monitoring.MetricServiceClient
+	conn   *grpc.ClientConn
+	ctx    context.Context
+	lis    *bufconn.Listener
+)
+
+func setup() {
```

Comment:
We should probably move some of these methods to a shared test util file since all tests will use them, but we can do that in a later PR